### ISSUE TITLE
Compilation error - Incorrectly extends interface Error

### ISSUE THREAD

**Issue by selvassn** [DataDog/browser-sdk#2442]
While building the angular app, it give the error as Interface 'ErrorWithCause' incorrectly extends interface Error

![image](https://github.com/DataDog/browser-sdk/assets/13289680/89428d9d-d13f-4362-8c14-34b44f14fd52)

Below is the Package.json

 "@angular/core": "~13.3.11",
    "@apollo/client": "3.3.21",
    "@datadog/browser-logs": "^4.50.0",
    "@datadog/browser-rum": "^4.50.0",
    "@datadog/datadog-ci": "^2.21.0",

**Comment by yannickadam** [comment]
Hi @selvassn,

I was able to reproduce your issue by changing my lib to `es2022` in `lib` of `tsconfig.json`, but I received a slightly different error message: 
```
Error: node_modules/@datadog/browser-core/cjs/domain/error/error.types.d.ts:2:18 - error TS2430: Interface 'ErrorWithCause' incorrectly extends interface 'Error'.
  Types of property 'cause' are incompatible.
    Type 'unknown' is not assignable to type 'Error | undefined'.
```

Could you provide us with more information about your environment (full package.json, tsconfig.json, ... ), or better a link to a small repo so we can reproduce the exact error message you are receiving? 
 
In the meantime, you should be able to fix this issue by adding: `skipLibCheck: true` to the `compilerOptions` section of your tsconfig.json.

**Comment by selvassn** [comment]
Hi , Thanks @yannickadam , below you can find tsconfig and package.json

 **tsconfig.json**

`{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "esnext",
      "es2018",
      "dom",
      "esnext.asynciterable"
    ]
  },
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true,
    "strictInjectionParameters": true,
  }
}`

**Package.json**

`"dependencies": {
    "@angular/animations": "~13.3.11",
    "@angular/cdk": "^13.0.0",
    "@angular/common": "~13.3.11",
    "@angular/compiler": "~13.3.11",
    "@angular/core": "~13.3.11",
    "@angular/flex-layout": "^13.0.0-beta.38",
    "@angular/forms": "~13.3.11",
    "@angular/localize": "~13.3.11",
    "@angular/material": "^13.3.9",
    "@angular/material-moment-adapter": "^13.3.9",
    "@angular/platform-browser": "~13.3.11",
    "@angular/platform-browser-dynamic": "~13.3.11",
    "@angular/router": "~13.3.11",
    "@apollo/client": "3.3.21",
    "@datadog/browser-logs": "^4.50.0",
    "@datadog/browser-rum": "^4.50.0",
    "@datadog/datadog-ci": "^2.21.0",
    "@fortawesome/fontawesome-free": "^5.15.4",
    "@ng-bootstrap/ng-bootstrap": "^11.0.0",
    "@ngx-formly/bootstrap": "^6.1.8",
    "@ngx-formly/core": "^6.1.8",
    "@ngx-formly/material": "^6.1.8",
    "@ngx-translate/core": "^13.0.0",
    "@ngx-translate/http-loader": "^6.0.0",
    "@types/jquery": "^3.5.1",
    "angular-bootstrap-md": "^13.0.0",
    "angular-rename": "^1.0.7",
    "apollo-angular": "3.0.1",
    "billboard.js": "^3.6.3",
    "bootstrap": "^4.5.2",
    "classlist.js": "^1.1.20150312",
    "d3": "^7.8.0",
    "graphql": "^15.5.1",
    "jquery": "^3.5.1",
    "json-logic-js": "^2.0.2",
    "jwt-decode": "^3.1.2",
    "launchdarkly-js-client-sdk": "^2.21.0",
    "moment": "^2.29.0",
    "ng2-charts": "3.1.2",
    "chart.js": "^3.9.1", 
    "rxjs": "^6.5.5",
    "save": "^2.4.0",
    "tslib": "^2.0.0",
    "wait-on": "^6.0.1",
    "web-animations-js": "^2.3.2",
    "zone.js": "~0.11.8"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^13.3.9",
    "@angular/cli": "^13.3.9",
    "@angular/compiler-cli": "~13.3.11",
    "@ngx-formly/schematics": "^6.1.8",
    "@types/jasmine": "~3.6.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^12.11.1",
    "codelyzer": "^6.0.0",
    "concurrently": "^5.3.0",
    "cypress": "^10.3.0",
    "cypress-real-events": "^1.7.1",
    "jasmine-core": "~3.8.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~6.3.20",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.7.0",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~4.6.4"
  }`

**Comment by selvassn** [comment]
```skipLibCheck: true``` works fine

**Comment by bcaudan** [comment]
Hello,

A fix for this issue has been released with browser-sdk@4.50.1.
Let us know if there is still an issue.
